From: Ubuntu Developers <ubuntu-devel-discuss@lists.ubuntu.com>
Date: Fri, 3 Apr 2026 15:15:13 +0800
Subject: fix-pam_motd-ftbfs

Patch for Ubuntu to fix FTBFS on oracular

Change in display_legal() of pam_motd.c as pam_get_item() expects third
argument as 'const void **'.

Authors: Nishit Majithia <nishit.majithia@canonical.com>

Upstream status: Ubuntu-specific
---
 modules/pam_motd/pam_motd.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/modules/pam_motd/pam_motd.c b/modules/pam_motd/pam_motd.c
index 59953aa..a9ee7c1 100644
--- a/modules/pam_motd/pam_motd.c
+++ b/modules/pam_motd/pam_motd.c
@@ -393,7 +393,7 @@ int display_legal(pam_handle_t *pamh)
     struct stat s;
     int f;
     /* Get the user name to determine if we need to print the disclaimer */
-    rc = pam_get_item(pamh, PAM_USER, &user);
+    rc = pam_get_item(pamh, PAM_USER, (const void **)&user);
     if (rc == PAM_SUCCESS && user != NULL && *(const char *)user != '\0')
     {
         PAM_MODUTIL_DEF_PRIVS(privs);
